### PR TITLE
Migrate iou_utils to the keras_cv.bounding_box namespace

### LINKED ISSUE #605: Migrate iou_utils to the `keras_cv.bounding_box` namespace
keras_cv/utils/iou.py => should be migrated to `keras_cv.bounding_box` as a user facing utility.

The utility will also have a required `bounding_box_format` argument.

we'll also need to update the COCO metrics to use this utility.

### PR DESCRIPTION
**Pull request by quantumalaviya**
# What does this PR do?

This adds support for the bounding box API.

fixes #605 

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.
@LukeWood

### CODE REVIEW

**Comment by quantumalaviya** [comment]
Moreover, it may be a good idea for this util to support the bounding box API.

@LukeWood What do you think?

**Comment by LukeWood** [comment]
A better solution to this would be to make this utility take a `bounding_box_format` argument.

**Comment by quantumalaviya** [comment]
@LukeWood updated this PR to fix #605

File: keras_cv/bounding_box/iou.py [hunk 1]
```diff
@@ -14,27 +14,40 @@
...
-    For example, the bounding box with it's bounding_box.TOP side at 100,
-    bounding_box.LEFT side at 200, bounding_box.BOTTOM at 101, and bounding_box.RIGHT
-    at 201 would be represented as:
-    > [100, 200, 101, 201]
+    The lookup vector is to be indexed by [`boxes1_index`,`boxes2_index`].
 
     Args:
       boxes1: a list of bounding boxes in 'corners' format.
       boxes2: a list of bounding boxes in 'corners' format.
+      bounding_box_format: a case-sensitive string which is one of `"xyxy"`,
```
Comment (LukeWood) [inline keras_cv/bounding_box/iou.py:28]: case-insensitive*

File: keras_cv/bounding_box/iou.py [hunk 2]
```diff
@@ -14,27 +14,40 @@
...
+        `"rel_xyxy"`, `"xyWH"`, `"center_xyWH"`, `"yxyx"`, `"rel_yxyx"`.
+        For detailed information on the supported format, see the
+        [KerasCV bounding box documentation](https://keras.io/api/keras_cv/bounding_box/formats/).
 
     Returns:
       iou_lookup_table: a vector containing the pairwise ious of boxes1 and
         boxes2.
     """
+
+    if "rel" in bounding_box_format:
```
Comment (LukeWood) [inline keras_cv/bounding_box/iou.py:38]: a comment explaining this would be helpful to future readers

**Review by LukeWood — CHANGES_REQUESTED** [review]
minor changes, but looks good to me otherwise!

File: keras_cv/bounding_box/iou.py (hunk 1 above)
Comment (quantumalaviya) [reply keras_cv/bounding_box/iou.py]: fixed this for nms as well!

File: keras_cv/bounding_box/iou.py [hunk 3]
```diff
@@ -14,27 +14,41 @@
 """Contains functions to compute ious of bounding boxes."""
 import tensorflow as tf
 
+from keras_cv import bounding_box
 
-def compute_ious_for_image(boxes1, boxes2):
-    """computes a lookup table vector containing the ious for a given set boxes.
 
-    The lookup vector is to be indexed by [`boxes1_index`,`boxes2_index`].
+def compute_ious_for_image(boxes1, boxes2, bounding_box_format):
```
Comment (LukeWood) [inline keras_cv/bounding_box/iou.py:20]: let's rename this to `compute_iou()`

let's also introduce batched support to this now that this is a utility.  Feel free to do batched support using map_fn, performance is not a huge concern for batched ATM, mainly concerned about API consistency.

**Review by LukeWood — CHANGES_REQUESTED** [review]
minor change, apologies for late changes

**Pull request merged**